internal/streaming/clickhouse_connector/connector.go

```diff
@@ -0,0 +1,201 @@
+package clickhouse_connector
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ClickHouse/clickhouse-go/v2"
+	"golang.org/x/exp/slog"
+
+	"github.com/openmeterio/openmeter/internal/sink"
+	"github.com/openmeterio/openmeter/internal/streaming"
+	"github.com/openmeterio/openmeter/pkg/models"
+)
+
+var prefix = "om"
+var eventsTableName = "events"
+
+// ClickhouseConnector implements `ingest.Connector“ and `namespace.Handler interfaces.
+type ClickhouseConnector struct {
+	config *ClickhouseConnectorConfig
+}
+
+type ClickhouseConnectorConfig struct {
+	Logger       *slog.Logger
+	KafkaConnect *sink.KafkaConnect
+	ClickHouse   clickhouse.Conn
+	Database     string
+}
+
+func NewClickhouseConnector(config *ClickhouseConnectorConfig) (*ClickhouseConnector, error) {
+	connector := &ClickhouseConnector{
+		config: config,
+	}
+
+	return connector, nil
+}
+
+func (c *ClickhouseConnector) Init(meter *models.Meter, namespace string) error {
+	// TODO: pass context to Init, also consider renaming it to CreateMeter
+	ctx := context.Background()
```

Comment:
Use context.TODO() so we know it needs to be changed (that's why it exists actually)